pkg/karmadactl/cmdinit/kubernetes/deploy.go

```diff
@@ -159,6 +168,10 @@ func (i *CommandInitOption) Complete() error {
 		}
 	}
 
+	if i.KarmadaAPIServerAdvertiseAddress != "" {
+		i.KarmadaAPIServerIP = append(i.KarmadaAPIServerIP, utils.StringToNetIP(i.KarmadaAPIServerAdvertiseAddress))
+	}
```

Comment:
In the case of `--karmada-apiserver-advertise-address` is set, do we still need to get IP from the master node?

Reply:
nice finding 
I will modify it tomorrow

Reply:
Maybe we can change `getKubeMasterIP` to something like `getKarmadaAPIServerIP`, then the `gocyclo` issue could be avoid as well.

Reply:
sounds better